src/grt/src/fastroute/src/RipUp.cpp

```diff
@@ -443,6 +443,53 @@ bool FastRouteCore::newRipup3DType3(const int netID, const int edgeID)
   return true;
 }
 
+void FastRouteCore::releaseResourcesNet(const int netID)
+{
+  const int edgeCost = nets_[netID]->edgeCost;
+  const TreeEdge* treeedges = sttrees_[netID].edges;
+  const int deg = sttrees_[netID].deg;
+
+  for (int edgeID = 0; edgeID < 2 * deg - 3; edgeID++) {
+
+    const TreeEdge* treeedge = &(treeedges[edgeID]);
+    const std::vector<short>& gridsX = treeedge->route.gridsX;
+    const std::vector<short>& gridsY = treeedge->route.gridsY;
+    const std::vector<short>& gridsL = treeedge->route.gridsL;
+    const int routeLen = treeedge->route.routelen;
+
+    for (int i = 0; i < routeLen; i++) {
+
+      if(gridsL[i] != gridsL[i + 1])
+        continue;
+
+      else if (gridsX[i] == gridsX[i + 1]) {  // a vertical edge
+
+        const int ymin = std::min(gridsY[i], gridsY[i + 1]);
+        if (v_edges_[ymin][gridsX[i]].usage >= edgeCost)
```

Comment:
Is it an error if the usage doesn't exceed the edgeCost?

Reply:
When the usage does not exceed the edgeCost and we subtract it, it becomes (~2^16), this is because the usage variable is unsigned.

Reply:
When is usage less than edgeCost?

Reply:
Please answer this question as this seems like a bug that usage would be less than the cost of a routed edge.

Reply:
There were parts of the code that were not updating the usage value, I updated and fixed the bug.